client/src/index.jsx

```diff
@@ -0,0 +1,221 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Price from './components/Price.jsx'
+import Photos from './components/Photos.jsx'
+import Description from './components/Description.jsx'
+// import Options from './components/Options.jsx'
+import Company from './components/Company.jsx'
+import Details from './components/Details.jsx'
+import Modal from './components/Modal.jsx'
+import AddToCart from './components/AddToCart.jsx'
+import AddToList from './components/AddToList.jsx'
+import ShareButtons from './components/ShareButtons.jsx'
+import MainPhoto from './components/MainPhoto.jsx'
+import QuantityMenu from './components/QuantityMenu.jsx'
+import StarRatingComponent from 'react-star-rating-component';
+import axios from 'axios';
+import primeLogo from '../dist/primelogo.png'
+ 
+class App extends React.Component {
+    constructor() {
+      super();
+      this.state = {
+        answers: 0,
+        description: '',
+        isHovered: false,
+        modalOpened: false,
+        showingModal: false,
+        photos: [],
+        prime: false,
+        mainphoto: '',
+        rating: 4.5,
+        selectedoption: '',
+        selectValue: '',
+        title: ''
+      }
+      this.getPhotos = this.getPhotos.bind(this);
+      this.hoverExit = this.hoverExit.bind(this);
+      this.hoverHandler = this.hoverHandler.bind(this);
+      this.handleModal = this.handleModal.bind(this);
+      this.hoverOptionHandler = this.hoverOptionHandler.bind(this);
+      this.handleQuantityChange = this.handleQuantityChange.bind(this);
+      this.showModal = this.showModal.bind(this);
+      this.hideModal = this.hideModal.bind(this);
+    }
+  
+    renderView() {
+    }
+
+    getMainProducts() {
+      let test = Math.floor(Math.random() * 10);
+      axios.get(`/api/mainProduct/${test}`).then((data) => {
+        let mainProductData = data.data;
+        console.log(mainProductData, 'line 48')
+        this.setState({
+          answers: mainProductData[0].answers,
+          company: mainProductData[0].company,
+          description: mainProductData[0].description,
+          price: mainProductData[0].price,
+          prime: mainProductData[0].prime,
+          ratings: mainProductData[0].ratings,
+          title: mainProductData[0].title
+        })
+      })
+    }
+
+    getPhotos() {
+      let test = Math.floor(Math.random() * 10);
+      axios.get(`/api/photos/${test}`).then((data) => {
+        let photourldata = data.data;
+        console.log(photourldata, 'line 52')
+        let photourlarray = [];
+        // TODO: access string inside object
+        photourldata.forEach(element => {photourlarray.push(element.photo_url)
+        console.log(photourlarray, 'logging data from photourldata line 26')
+        });
+        // TODO: refactor filter method of getting correct urls
+          this.setState({
```

Comment:
storing an array of only the url might make it hard to add features in the future, IE if you had different product options you wanted to display. Would probably make more sense to store an array of photo objects and work from there.